sklearn/neighbors/_nearest_centroid.py

```diff
@@ -134,6 +141,20 @@ def fit(self, X, y):
             Fitted estimator.
         """
         self._validate_params()
+
+        if isinstance(self.metric, str) and self.metric not in (
+            "manhattan",
+            "euclidean",
+        ):
+            warnings.warn(
+                (
+                    "Support for metrics other than euclidean and "
```

Comment:
Distance metrics and metrics are two different concepts. I think we need to be clear despite the potentially misleading name of this parameter.

```suggestion
                    "Support for other distance metrics than euclidean and "
```